kong/plugins/tcp-log/handler.lua

```diff
@@ -27,13 +27,15 @@ local function log(premature, conf, message)
   local ok, err = sock:connect(host, port)
   if not ok then
     kong.log.err("failed to connect to ", host, ":", tostring(port), ": ", err)
+    sock:close()
```

Comment:
I'm a little curious, is it still need to explicitly call `close` when an error occurs? Most languages don't seem to need to do that, and it doesn't seem to do so in the official [examples](https://github.com/openresty/lua-nginx-module#tcpsockconnect) I've seen.

Reply:
A reasonable concern. Please let us know if you have some conclusion about it. @oowl

Reply:
> Fatal errors in cosocket operations always automatically close the current connection (note that, read timeout error is the only error that is not fatal), and if you call [close](https://github.com/openresty/lua-nginx-module#tcpsockclose) on a closed connection, you will get the "closed" error.

https://github.com/openresty/lua-nginx-module#ngxsockettcp

Even if the socket will be closed when a fatal occurred, I think closing a failed socket explicitly is a good style.

Reply:
> I think closing a failed socket explicitly is a good style.

I have the opposite opinion, I think such code is not clean enough and feels very verbose.